go/heif/heif.go

```diff
@@ -312,41 +312,49 @@ func (c *Context) ReadFromFile(filename string) error {
 	defer C.free(unsafe.Pointer(c_filename))
 
 	err := C.heif_context_read_from_file(c.context, c_filename, nil)
+	runtime.KeepAlive(c)
 	return convertHeifError(err)
 }
 
 func (c *Context) ReadFromMemory(data []byte) error {
 	// TODO: Use reader API internally.
 	err := C.heif_context_read_from_memory(c.context, unsafe.Pointer(&data[0]), C.size_t(len(data)), nil)
+	runtime.KeepAlive(c)
 	return convertHeifError(err)
 }
 
 func (c *Context) GetNumberOfTopLevelImages() int {
-	return int(C.heif_context_get_number_of_top_level_images(c.context))
+	i := int(C.heif_context_get_number_of_top_level_images(c.context))
+	runtime.KeepAlive(c)
+	return i
 }
 
 func (c *Context) IsTopLevelImageID(ID int) bool {
-	return C.heif_context_is_top_level_image_ID(c.context, C.heif_item_id(ID)) != 0
+	ok := C.heif_context_is_top_level_image_ID(c.context, C.heif_item_id(ID)) != 0
+	runtime.KeepAlive(c)
+	return ok
 }
 
 func (c *Context) GetListOfTopLevelImageIDs() []int {
 	num := int(C.heif_context_get_number_of_top_level_images(c.context))
+	runtime.KeepAlive(c) // TODO: Is this really neccessary? Will be called again in the same func.
```

Comment:
As discussed offline, these calls will stay to avoid the same problems we're currently facing if the call to `runtime.KeepAlive` further down below gets refactored away (which would then not call `runtime.KeepAlive` at all in this func).